### PR TITLE
containers: Fix "use of untyped nil"

### PR DESCRIPTION
**Pull request by abhinav**
When parsing containers with primitives, the `fromWire` expression is,
`value.Get$Type(), nil` (where `$Type` is the primitive type). While reading
an item from Value, this becomes,

    foo, err := value.GetI32(), nil

Assigning nil to a variable with no type information will result in "use of
untyped nil". To fix this, we need to add, `var err error` on the line before
it.

    var err error
    foo, err := value.GetI32(), nil

This changes the code generator for lists, sets, and maps to do this.

Also added tests for containers containing primitive types.

### CODE REVIEW

**Comment by abhinav** [comment]
@breerly @akshayjshah @prashantv

File: gen/container_test.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+						{Key: wire.NewValueString("foo"), Value: wire.NewValueBool(true)},
+						{Key: wire.NewValueString("bar"), Value: wire.NewValueBool(false)},
+						{Key: wire.NewValueString("baz"), Value: wire.NewValueBool(true)},
+					}),
+				}),
+			}}}),
+		},
+	}
+
+	for _, tt := range tests {
```
Comment (breerly) [inline gen/container_test.go]: that test table though :)

**Comment by breerly** [comment]
:sheep: :it:

**Comment by prashantv** [comment]
lgtm

**Pull request merged**